frontend/src/core/entities/selectors.ts

```diff
@@ -8,21 +8,22 @@ import { NAME } from '.';
 import type { Entities, Entity } from 'core/api';
 import type { NavigationParams } from 'core/navigation';
 import type { UserState } from 'core/user';
+import type { RootState } from '../../store';
 
-const entitiesSelector = (state): Entities => state[NAME].entities;
-const userSelector = (state): UserState => state[user.NAME];
+const entitiesSelector = (state: RootState) => state[NAME].entities;
+const userSelector = (state: RootState) => state[user.NAME];
 
 export function _getSelectedEntity(
-    entities: Entities,
+    entities: ReturnType<typeof entitiesSelector>,
     params: NavigationParams,
-): Entity | null | undefined {
+): Entity | undefined {
```

Comment:
I wonder what kinda state we're in if this doesn't have an Entity?

There's a bunch of code in the frontend that assumes this exists.

Reply:
A very state. I _think_ it's unreachable unless the state or the store get horribly corrupted somehow.

Really this might be taken as a sign that we should deeply refactor the state handling to reflect this. Not sure if it's worth the trouble, though.